torchgeo/datasets/xview.py

```diff
@@ -0,0 +1,390 @@
+            image1 = os.path.join(image_root, f"{name}_pre_disaster.png")
+            image2 = os.path.join(image_root, f"{name}_post_disaster.png")
+            mask1 = os.path.join(mask_root, f"{name}_pre_disaster_target.png")
+            mask2 = os.path.join(mask_root, f"{name}_post_disaster_target.png")
+            files.append(dict(image1=image1, image2=image2, mask1=mask1, mask2=mask2))
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the target mask
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("L"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            tensor = tensor.to(torch.long)  # type: ignore[attr-defined]
+            return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if checksum fails or the dataset is not downloaded
+        """
+        # Check if the files already exist
+        exists = []
+        for split_info in self.metadata.values():
+            for directory in ["images", "labels", "targets"]:
+                exists.append(
+                    os.path.exists(
+                        os.path.join(self.root, split_info["directory"], directory)
+                    )
+                )
+
+        if all(exists):
+            return
+
+        # Check if .tar.gz files already exists (if so then extract)
+        exists = []
+        for split_info in self.metadata.values():
+            filepath = os.path.join(self.root, split_info["filename"])
+            if os.path.isfile(filepath):
+                if self.checksum and not check_integrity(filepath, split_info["md5"]):
+                    raise RuntimeError("Dataset found, but corrupted.")
+                exists.append(True)
+                extract_archive(filepath)
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the user requested to download the dataset
+        raise RuntimeError(
+            "Dataset not found in `root` directory, either specify a different"
+            + " `root` directory or manually download the dataset to this directory."
+        )
+
+    @staticmethod
```

Comment:
@calebrob6 I don't think we want these to be static methods in general. I can imagine a dataset like chesapeake or so2sat where the cmap depends on the specific band set a user has chosen when instantiating a dataset. This information can be acquired from `self` if it's a regular class method.

Reply:
yeah great point